### DISCUSSION TITLE
Crate for "system directories"?

### DISCUSSION THREAD

**Discussion by alexveecle** [nicoburns/blessed-rs#139] [Ideas]
It could be cool to bless some library to obtain the user's cache dir, config dir for your app, etc.- this helps write nice CLI tools.

I have seen https://docs.rs/dirs/latest/dirs/ and many others, although I have no experience with any of them (I came here to see if something was recommended already).

**Comment by djc** [comment]
It's a good question -- there's been a fair bit of churn in these crates.

* dirs has 7.8M recent downloads, last update over 1 year ago
* directories has 2.6M recent downloads, last update over 1 year ago
* There's also the home crate which is maintained by the Cargo team, and is more minimal, with 16M recent downloads

Would you be able to put up a PR with these?

**Reply by alexveecle** [reply]
Unfortunately, I think https://crates.io/crates/home is excessively minimalistic- I assume it solves the problem it wants to solve for `cargo`, but really I think most applications should not use `~/.app`. And I don't feel qualified to choose between dirs and directories. Are you suggesting a PR that suggests both crates?

**Reply by djc** [reply]
Yes, I think it makes sense to list both, with `dirs` going first on account of it being more popular. Ideally we'd provide some notes on the difference but I think it would be okay to punt on that for now.

**Reply by nicoburns** [reply]
`dirs` and `directories` seem to be very similar crates maintained by the same organisation, which `dirs` being slightly lower level and `directories` being slightly higher level. Seems reasonable to mention both. Could even just be one entry (probably for `dirs` with `directories` mentioned in the notes (HTML links work in notes sections)).

There seems to have been some [mild controversy](https://github.com/dirs-dev/dirs-sys-rs/issues/21) about the introduction of a (first-party) MPL dependency into the dependency tree of both crates, with some people switching to `etcetera` or custom code (which is built on top of home and has a similar number of downloads to `directories`) for this reason. Could be worth mentioning as a 3rd option, although we'll want to be careful how we phrase it to avoid stoking conflict.

I think `dirs` should probably still be the primary recommendation.

**Reply by alexveecle** [reply]
Tried https://github.com/nicoburns/blessed-rs/pull/142 ; I think this can be marked as resolved and any discussion can continue there.
